### ISSUE TITLE
Incorrect warnings when mixing an ordered parent list with an unordered child list

### ISSUE THREAD

**Issue by KyleMit** [DavidAnson/vscode-markdownlint#121]
I believe this is unintended warning of MD032, MD007, & MD029 when mixing an ordered parent list with an unordered child list

Repro:

```none
1. Go to Store and get
  * eggs
  * bread
2. Fix dishwasher
```

Here's the warnings that get generated from this snippet in VS Code:

![example screenshot](https://user-images.githubusercontent.com/4307307/86841713-4682bf80-c072-11ea-8476-e836d5f3f9d1.png)

> **MD032** / blanks-around-lists: Lists should be surrounded by blank lines
> **MD007** / ul-indent: Unordered list indentation [Expected: 0; Actual: 2]
> **MD032** / blanks-around-lists: Lists should be surrounded by blank lines
> **MD007** / ul-indent: Unordered list indentation [Expected: 0; Actual: 2]
> **MD032** / blanks-around-lists: Lists should be surrounded by blank lines
> **MD029** / ol-prefix: Ordered list item prefix [Expected: 1; Actual: 2; Style: 1/1/1]
> **MD032** / blanks-around-lists: Lists should be surrounded by blank lines

If we compare to other configurations of nested lists, all other parent child relationships seem to work okay with 2 spaces

```none
## Unordered > Unordered ✔

* Parent
  * Child

## Ordered > Ordered ✔

1. Parent
   1. Child

## Unordered > Ordered ✔

* Parent
  1. Child

## Ordered > Unordered ❌

1. Parent
  * Child
```

This is true with and without the markdownlint config set to 2 indentation spaces as follows:

```json
"markdownlint.config": {
    "MD007": { "indent": 2 },
},
```

**Workaround**: the errors go away if you add a third space to child items, but this doesn't seem like the right long term approach

**Comment by DavidAnson** [comment]
The scenario you show at the top is explicitly wrong according to the CommonMark specification, specifically Examples [266](https://spec.commonmark.org/0.29/#example-266) and [267](https://spec.commonmark.org/0.29/#example-267). Section 5.2.1 of the spec goes into more detail about this requirement.

Tour example renders as [three different lists](https://dlaa.me/markdownlint/#%25m%23%20Issue%20121%0A%0A1.%20Go%20to%20Store%20and%20get%0A%20%20*%20eggs%0A%20%20*%20bread%0A2.%20Fix%20dishwasher%0A) with a CommonMark-compliant parser which is not what you intend.

As you confirm, using [three spaces to indent the sublist](https://dlaa.me/markdownlint/#%25m%23%20Issue%20121%0A%0A1.%20Go%20to%20Store%20and%20get%0A%20%20%20*%20eggs%0A%20%20%20*%20bread%0A2.%20Fix%20dishwasher%0A) renders as you intend and reports no issues.

I think the current behavior is correct and helpful, do you disagree?

**Comment by KyleMit** [comment]
My mistake - thank you for pointing out the spec, and how it illustrates the correct behavior.  I should've done more due diligence first.  I attempted to resolve with the indentation count, thinking that I'd have to sick to 2 spaces for all levels of indentation, but missed the need for two character numbered lists to receive special treatment.

Appreciate the library and the answer.

**Comment by DavidAnson** [comment]
No worries! This is an area of frequent discussion. :)
